python_ta/typecheck/base.py

```diff
@@ -389,31 +389,42 @@ def find_parent(self, tn: _TNode, find_repr: bool = False) -> Optional[_TNode]:
         """Do a bfs starting from tn to find a _TNode that has a parent."""
         if tn.parent:
             return tn.parent
+        goal_tnode = self.find_node(tn, (lambda t: t.parent), find_repr)
+
+        if goal_tnode and goal_tnode.parent:
+            cur_node = goal_tnode
+            while cur_node:
+                next_node = None
+                for e in cur_node.adj_list:
+                    if not e[0].parent_path and not e[0].parent:
+                        e[0].parent = goal_tnode
+                        e[0].parent_path = (cur_node, e[1])
+                        next_node = e[0]
+                cur_node = next_node
+
+        return goal_tnode
+
+    def find_function_def(self, tn: _TNode) -> astroid.FunctionDef:
```

Comment:
This is a bit funky. Better to make the return type `Optional` (and above as well), but it's more noticeable here.